core/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java

```diff
@@ -202,4 +221,29 @@ public int hashCode() {
         result = 31 * result + (unit != null ? unit.hashCode() : 0);
         return result;
     }
+
+    @Override
+    public int compareTo(ByteSizeValue other) {
+        long unitValue = unit.toBytes(1);
+        long otherUnitValue = other.unit.toBytes(1);
+
+        ByteSizeUnit minUnit = unitValue < otherUnitValue ? unit : other.unit;
+        long thisValue = this.convert(minUnit);
+        long otherValue = other.convert(minUnit);
+
+        if (thisValue == otherValue) {
```

Comment:
I had it like that originally, but changed it since if we convert to Bytes and compare, every value above (Long.MAX_LONG, BYTES) will be treated as equals. 

Is there a reason why we want this behavior? (I noticed the writeTo method also writes getBytes())

For the _cat api, I think it uses ByteSizeValue directly in some cases to support the ability to change numeric formats, e.g. the ?bytes=b thing.